lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -96,6 +101,10 @@ public class BuildCriOauthRequestHandler
     private final ClientOAuthSessionDetailsService clientOAuthSessionDetailsService;
     private final Gpg45ProfileEvaluator gpg45ProfileEvaluator;
     private final SessionCredentialsService sessionCredentialsService;
+    private final OAuthKeyService oAuthKeyService;
+
+    private static final Integer DEFAULT_OAUTH_CRI_ENC_KEY_CACHE_DURATION_MINUTES = 5;
+    private static Map<String, CachedOAuthCriEncryptionKey> cachedOAuthCriEncryptionKeys;
```

Comment:
I think we should move the caching logic into the `OauthKeyService` as well - this class shouldn't really have to worry about cache implementation details.